orchestrator/api/workspace_github.py

```
@@ -0,0 +1,205 @@
+    page: int = Query(1, ge=1),
+    per_page: int = Query(30, ge=1, le=100),
+    ctx: RequestContext = Depends(get_request_context_hybrid),
+):
+    """List GitHub repos accessible to the user via Composio."""
+    if str(ctx.workspace_id) != workspace_id:
+        raise HTTPException(status_code=403, detail="Workspace access denied")
+
+    try:
+        from core.composio.client import get_composio_client
+    except ImportError:
+        raise HTTPException(status_code=501, detail="Composio SDK not installed")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)